app_dart/lib/src/model/ci_yaml/target.dart

```diff
@@ -40,12 +40,30 @@ class Target {
 
   static String kIgnoreFlakiness = 'ignore_flakiness';
 
-  /// Gets dimensions for this [pb.Target].
+  /// Gets assembled dimensions for this [pb.Target].
   ///
   /// Swarming dimension doc: https://chromium.googlesource.com/infra/luci/luci-go/+/HEAD/lucicfg/doc/README.md#swarming.dimension
+  ///
+  /// Target dimensions are prioritized in:
+  ///   1. [schedulerConfig.platformDimensions]
+  ///   2. [pb.Target.properties]
   List<RequestedDimension> getDimensions() {
-    final Map<String, Object> properties = getProperties();
     final List<RequestedDimension> dimensions = <RequestedDimension>[];
+
+    final Map<String, Object> platformDimensions = _getPlatformDimensions();
+    for (String key in platformDimensions.keys) {
+      String value = platformDimensions[key].toString();
+      dimensions.add(RequestedDimension(key: key, value: value));
+    }
+
+    final Map<String, Object> targetDimensions = _getTargetDimensions();
+    for (String key in targetDimensions.keys) {
+      String value = targetDimensions[key].toString();
+      dimensions.add(RequestedDimension(key: key, value: value));
+    }
+
+    final Map<String, Object> properties = getProperties();
+
     for (String dimension in dimensionList) {
```

Comment:
We should deprecate dimensions defined in properties to make the logic more clear. With that, this would not be needed any more.

To remove this logic, we need to make sure .ci.yaml have been updated to move `dimension` values under `properties` to the newly supported `dimensions` section.

Let's add a TODO to remove this logic when .ci.yaml are updated.

Reply:
Nit: a TODO comes with a bug created to track.